### PR TITLE
Enable OPTIONS preflight and a method to enable cros origins

### PR DESCRIPTION
**Pull request by giurgiur99**
**Reasoning**
- At the moment, no support for OPTIONS request is implemented. This is the preflight request for CORS checks, thus API calls to a serverless function from a web browser is failing.

**Changes**
- Enable preflight response **[OPTIONS]** with generic cors allow.
- Export a function "cors" where you can specify the allowed origins.

### CODE REVIEW

**Comment by lholmquist** [comment]
tests are failing

File: index.js [hunk 1]
```diff
@@ -136,7 +139,7 @@ function initializeServer(config) {
   // Add a parser for application/x-www-form-urlencoded
   server.addContentTypeParser(
     'application/x-www-form-urlencoded',
-    function(_, payload, done) {
+    function (_, payload, done) {
```
Comment (lholmquist) [inline index.js:142]: unrelated formatting

File: index.js [hunk 2]
```diff
@@ -156,7 +159,7 @@ function initializeServer(config) {
   server.addContentTypeParser(
     '*',
     { parseAs: 'buffer' },
-    function(req, body, done) {
+    function (req, body, done) {
```
Comment (lholmquist) [inline index.js:162]: unrelated formatting

File: lib/invoker.js [hunk 3]
```diff
@@ -3,20 +3,28 @@
...
     // Default payload values
     const payload = {
       code: 200,
       response: undefined,
       headers: {
         'content-type': 'application/json; charset=utf-8',
-        'access-control-allow-methods':
-          'OPTIONS, GET, DELETE, POST, PUT, HEAD, PATCH',
-        'access-control-allow-origin': '*'
+        'access-control-allow-methods': 'OPTIONS, GET, POST',
```
Comment (lholmquist) [inline lib/invoker.js:15]: why are you removing the other methods(DELETE, PUT, HEAD, PATCH)?  This should stay the same since it is the default, otherwise it would be a breaking change

Comment (giurgiur99) [reply lib/invoker.js]: **invocation-handler.js**

```javascript
module.exports = function use(fastify, opts, done) {
  fastify.get('/', doGet);
  fastify.post('/', doPost);
  fastify.options('/', doOptions);
  const invokeFunction = invoker(opts);
```

As you can see here, only the post and get were supported and now i added options. Why should the rest of the methods be allowed if they cannot be used? Anyway, I added the rest of allowed methods in cors.

**Comment by giurgiur99** [comment]
@lholmquist I fixed the code, according to the tests and removed formatting differences. Can you take another look? Thanks

**Comment by lholmquist** [comment]
@giurgiur99 can you please add a test/tests for this new functionality?

**Comment by giurgiur99** [comment]
@lholmquist I added tests for preflight and allowed origins

**Review by lholmquist — APPROVED** [review]
looks good.  the ci is failing, but not because of this PR
